ctapipe/tools/make_irf.py

```diff
@@ -0,0 +1,342 @@
+"""Tool to generate IRFs"""
+import operator
+from pathlib import Path
+
+import astropy.units as u
+import numpy as np
+from astropy.io import fits
+from astropy.table import vstack
+from pyirf.benchmarks import angular_resolution, energy_bias_resolution
+from pyirf.binning import (
+    add_overflow_bins,
+    create_bins_per_decade,
+    create_histogram_table,
+)
+from pyirf.cut_optimization import optimize_gh_cut
+from pyirf.cuts import calculate_percentile_cut, evaluate_binned_cut
+from pyirf.io import (
+    create_aeff2d_hdu,
+    create_background_2d_hdu,
+    create_energy_dispersion_hdu,
+    create_psf_table_hdu,
+    create_rad_max_hdu,
+)
+from pyirf.irf import (
+    background_2d,
+    effective_area_per_energy,
+    energy_dispersion,
+    psf_table,
+)
+from pyirf.sensitivity import calculate_sensitivity, estimate_background
+from pyirf.simulations import SimulatedEventsInfo
+from pyirf.spectral import (
+    CRAB_HEGRA,
+    IRFDOC_ELECTRON_SPECTRUM,
+    IRFDOC_PROTON_SPECTRUM,
+    PowerLaw,
+    calculate_event_weights,
+)
+from pyirf.utils import calculate_source_fov_offset, calculate_theta
+
+from ..core import Provenance
+from ..io import TableLoader
+from ..irf import DataBinning, IrfToolBase
+
+PYIRF_SPECTRA = {
+    "CRAB_HEGRA": CRAB_HEGRA,
+    "IRFDOC_ELECTRON_SPECTRUM": IRFDOC_ELECTRON_SPECTRUM,
+    "IRFDOC_PROTON_SPECTRUM": IRFDOC_PROTON_SPECTRUM,
+}
+
+
+class IrfTool(IrfToolBase, DataBinning):
+    name = "ctapipe-make-irfs"
+    description = "Tool to create IRF files in GAD format"
+
+    def make_derived_columns(self, events, spectrum, target_spectrum):
+        events["pointing_az"] = 0 * u.deg
```

Comment:
This can be obtained nowadays from the observation block information (and is loaded if you pass `load_observation_info` to `TableLoader`.